app/assets/javascripts/components/stories/StorySearch.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import { search } from "actions/story";
+import { connect } from "react-redux";
+
+const searchDiv = document.getElementById('portal-search-input');
+
+export class StorySearch extends Component {
+  constructor(props) {
+    super(props);
+
+    const { projectId, search } = props;
+
+    this.state = {
+      keyWord: '',
+      searchedStories: [],
+      projectId,
+      search
+    }
+  }
+  
+  handleChange = e => {
+    this.setState({ keyWord: e.target.value })
+  }
+
+  handleSearch = async (e) => {
+    e.preventDefault()
+
+    const { keyWord, projectId, search } = this.state
+
+    search(keyWord, projectId)
+  }
+
+  render() {
+    const { keyWord } = this.state;
+
+    return ReactDOM.createPortal(
```

Comment:
I don't know how is the design pattern of this project but I would extract this portal for a new component:
```js
import ReactDOM from 'react-dom';

const PortalSearchInput = ({ children }) => ReactDOM.createPortal(
  children,
  document.getElementById('portal-search-input'),
);

export default PortalSearchInput;
```

Also, I sould put this in a folder called `portals`.